bot/bot.py

```diff
@@ -44,9 +45,23 @@ def __init__(self, *args, redis_session: RedisSession, **kwargs):
             # Since statsd is UDP, there are no errors for sending to a down port.
             # For this reason, setting the statsd host to 127.0.0.1 for development
             # will effectively disable stats.
-            statsd_url = "127.0.0.1"
+            statsd_url = LOCALHOST
 
-        self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        self.stats = AsyncStatsClient(self.loop, LOCALHOST)
+        self.connect_statsd(statsd_url)
+
+    def connect_statsd(self, statsd_url: str, retry_after: int = 2, attempt: int = 1) -> None:
+        """Callback used to retry a connection to statsd if it should fail."""
+        if attempt > 5:
+            log.error("Reached 10 attempts trying to reconnect AsyncStatsClient. Aborting")
+            return
+
+        try:
+            self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        except socket.gaierror:
+            log.warning(f"Statsd client failed to connect (Attempts: {attempt})")
+            # Use a fallback strategy for retrying, up to 5 times.
+            self.loop.call_later(retry_after, self.retry_statsd_connection, statsd_url, retry_after ** 2, attempt + 1)
```

Comment:
The sixth attempt will wait over 18 hours, which is too excessive. 5 attempts will only be a bit over 4 minutes. I don't think an exponential function is good, because waiting 4 minutes is not that long, but the next best is 18 hours, which is too long.